pay-api/src/pay_api/models/statement.py

```diff
@@ -115,14 +116,49 @@ def payment_methods(self):
 
         return list(payment_methods)
 
+    @hybrid_property
+    def amount_owing(self):
+        """Return amount owing on statement."""
+        from .statement_invoices import StatementInvoices  # pylint: disable=import-outside-toplevel
+        query = self.get_statement_owing_query()
+        query = (query.filter(StatementInvoices.statement_id == self.id)
+                 .with_entities(func.sum(Invoice.total - Invoice.paid).label('total_owing')))
+
+        result = query.scalar()
+        return result if result is not None else 0
+
+    @staticmethod
+    def get_statement_owing_query():
+        """Get statement query used for amount owing."""
+        from .statement_invoices import StatementInvoices  # pylint: disable=import-outside-toplevel
```

Comment:
not good

Reply:
Needed to use this as part of the query existing in the model, I originally tried to do this in the statement service, but 

service --> model --> service.

Should I just refactor the model and pull out the statement search up to the service?

Reply:
Yes